clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,398 @@
+                            config.log_level = log_level;
+                        }
+                    }
+                }
+                "log_file_path" => {
+                    if let Some(log_file_path) = value.as_str() {
+                        config.log_file_path = log_file_path.to_string();
+                    }
+                }
+                "authentication_token" => {
+                    if let Some(auth_token) = value.as_str() {
+                        config.authentication_token = auth_token.to_string();
+                    }
+                }
+                "app_user_uid" => {
+                    if let Some(uid_str) = value.as_str() {
+                        if let Ok(uid) = Uuid::parse_str(uid_str) {
+                            config.app_user_uid = uid;
+                        }
+                    }
+                }
+                "interface" => {
+                    if let Ok(interface) = serde_json::from_value::<InterfaceConfig>(value) {
+                        config.interface = interface;
+                    }
+                }
+                "http_certificate_file_path" => {
+                    if value.is_null() {
+                        config.http_certificate_file_path = None;
+                    } else if let Some(path_str) = value.as_str() {
+                        config.http_certificate_file_path = Some(PathBuf::from(path_str));
+                    }
+                }
+                "mqtt" => {
+                    if let Ok(mqtt_config) = serde_json::from_value::<MqttConfig>(value) {
+                        config.mqtt = mqtt_config;
+                    }
+                }
+                _ => {}
+            }
+        }
+    } else {
+        return text_response(400, "Invalid JSON format");
+    }
+
+    match fs::write(TELIOD_CFG, serde_json::to_string_pretty(&config).unwrap()) {
+        Ok(_) => text_response(200, "Configuration updated successfully"),
+        Err(_) => text_response(500, "Failed to write updated config"),
+    }
+}
+
+fn get_status() -> Response {
+    if is_teliod_running() && teliod_socket_exists() {
+        if let Ok(socket_path) = DaemonSocket::get_ipc_socket_path() {
+            let socket_query = futures::executor::block_on(tokio::time::timeout(
+                Duration::from_secs(TIMEOUT_SEC),
+                DaemonSocket::send_command(
+                    &socket_path,
+                    &serde_json::to_string(&ClientCmd::GetStatus).unwrap_or_default(),
+                ),
+            ));
+
+            if let Ok(Ok(daemon_reply)) = socket_query {
+                if let Ok(CommandResponse::StatusReport(status)) =
+                    CommandResponse::deserialize(&daemon_reply)
+                {
+                    return text_response(
+                        200,
+                        serde_json::to_string_pretty(&status).unwrap_or_default(),
+                    );
+                }
+            }
```

Comment:
Would be nice to know that some errors happened - please add some sort of logging or return response with a useful string that includes the error.

Reply:
added more specific error handling